jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -192,8 +190,26 @@ static String getPathInContext(Request request)
      */
     HttpFields getTrailers();
 
+    /**
+     * <p>Get the millisecond timestamp that the request was created, relative to {@link System#currentTimeMillis()}.
+     * This method should be used for wall clock time, rather than {@link #getNanoTimeStamp()},
+     * which is appropriate for measuring latencies.</p>
+     * @return The timestamp that the request was received/created in milliseconds, relative to the "wall clock" given
+     * by {@link System#currentTimeMillis()}.
+     */
     long getTimeStamp();
 
+    /**
+     * <p>Get the nanosecond timestamp that the request was created, relative to {@link System#nanoTime()}.
+     * This method should be used when measuring latencies, rather than {@link #getTimeStamp()},
+     * which is appropriate for wall clock time.</p>
+     * @return The timestamp that the request was received/created in nanoseconds, relative to {@link System#nanoTime()}.
+     */
+    default long getNanoTimeStamp()
+    {
+        return NanoTime.now() - TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis() - getTimeStamp());
```

Comment:
I don't think this implementation is correct, since `currentTimeMillis()` can drift.
Must use another field to store the creation/received nanoTime.

Reply:
@sbordet we do. It is in the HttpStream.  This is just used for test code.